app/models/username_validator.rb

```diff
@@ -65,21 +80,21 @@ def username_length_max?
 
   def username_char_valid?
     return unless errors.empty?
-    if username =~ /[^\w.-]/
+    unless username[/^[#{UNICODE_RANGES}\.\d\-_]+$/]
```

Comment:
Why not use `[[:alphanumeric:]]` instead?

Reply:
I guess you are talking of `/[[:alnum]]/` that seems like a good idea to me. We have ranges here because originally I was given a defined list of accepted unicode ranges. But that would probably work, will change the PR to use this and see if it creates issues.